android/src/main/java/com/snowplowanalytics/react/util/EventUtil.java

```diff
@@ -2,115 +2,200 @@
 
 import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.bridge.ReadableArray;
-import com.snowplowanalytics.snowplow.tracker.events.SelfDescribing;
-import com.snowplowanalytics.snowplow.tracker.events.Structured;
-import com.snowplowanalytics.snowplow.tracker.payload.SelfDescribingJson;
-import com.snowplowanalytics.snowplow.tracker.events.ScreenView;
-import com.snowplowanalytics.snowplow.tracker.events.PageView;
+import com.snowplowanalytics.snowplow.event.SelfDescribing;
+import com.snowplowanalytics.snowplow.event.Structured;
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson;
+import com.snowplowanalytics.snowplow.event.ScreenView;
+import com.snowplowanalytics.snowplow.event.PageView;
+import com.snowplowanalytics.snowplow.event.Timing;
+import com.snowplowanalytics.snowplow.event.ConsentGranted;
+import com.snowplowanalytics.snowplow.event.ConsentWithdrawn;
+import com.snowplowanalytics.snowplow.event.EcommerceTransactionItem;
+import com.snowplowanalytics.snowplow.event.EcommerceTransaction;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.UUID;
 
 public class EventUtil {
 
-    public static List<SelfDescribingJson> getContexts(ReadableArray contexts) {
+    public static SelfDescribingJson getSelfDescribingJson(ReadableMap json) {
+        String schema = json.getString("schema");
+        ReadableMap dataMap = json.getMap("data");
 
+        return new SelfDescribingJson(schema, dataMap.toHashMap());
+    }
+
+    public static List<SelfDescribingJson> getContexts(ReadableArray contexts) {
         List<SelfDescribingJson> nativeContexts = new ArrayList<>();
         for (int i = 0; i < contexts.size(); i++) {
-            SelfDescribingJson json = EventUtil.getSelfDescribingJson(contexts.getMap(i));
+            SelfDescribingJson json = getSelfDescribingJson(contexts.getMap(i));
             nativeContexts.add(json);
         }
+
         return nativeContexts;
     }
 
-    public static SelfDescribingJson getSelfDescribingJson(ReadableMap json) {
+    public static Structured getStructuredEvent(ReadableMap argmap) {
```

Comment:
nit: Should the methods in this file be `create...` or `build...` rather than `get...`? Feels like we'd constructing new objects rather than retrieving existing ones.